src/System.Private.CoreLib/shared/System/String.Manipulation.cs

```diff
@@ -195,22 +196,30 @@ public static string Concat<T>(IEnumerable<T> values)
                         return firstString ?? string.Empty;
                     }
 
-                    StringBuilder result = StringBuilderCache.Acquire();
+                    Span<char> initialBuffer = stackalloc char[64];
+                    var result = new ValueStringBuilder(initialBuffer);
 
-                    result.Append(firstString);
+                    if (firstString != null)
+                    {
+                        result.Append(firstString);
+                    }
 
                     do
                     {
                         currentValue = en.Current;
 
                         if (currentValue != null)
                         {
-                            result.Append(currentValue.ToString());
+                            string? str = currentValue.ToString();
```

Comment:
There is a chance that the `T` supports `ISpanFormattable`. Calling the object overload should avoid the additional string allocation in that case.

Reply:
Yes, it should be done in separate change.

Reply:
It's certainly possible, but I have to wonder how common it actually is.  Calling Concat (not Join) with a bunch of integers, floating-point values, Guids, DateTimes, and TimeSpans is going to result in a bunch of gobbledeegook, as there won't be any separators between the values.  Do you have an example where this would happen and be worth optimizing for?  My gut would be to avoid the interface check as it actually succeeding would be super rare.

Reply:
I'm thinking more of the future- in theory we'd want to make this interface public eventually, no?

Reply:
> I'm thinking more of the future

Even then, what's an example where this would be valuable? It would need to be a `T` where performance was important and where TryFormat included the necessary punctuation or spacing to make Concat a useful rendering.

In trying to find such examples, I'm not finding any... in fact I'm finding very little usage of this overload in general.

If you have such cases in mind, I'd be interested in seeing them.

Thanks.

Reply:
> It would need to be a T where performance is important ...

And then you probably wouldn't want to be using an `IEnumerable<T>` overload?